R/build.R

```diff
@@ -25,15 +25,20 @@
 #' @param compile_attributes if `TRUE` and the package uses Rcpp, call
 #'   [Rcpp::compileAttributes()] before building the package. It is ignored
 #'   if package does not need compilation.
+#' @param register_routines if `TRUE` and the package does not use Rcpp, call
+#'   register routines with
+#'   [tools::package_native_routine_registration_skeleton()] before building
+#'   the package. It is ignored if package does not need compilation.
 #' @export
 #' @return a string giving the location (including file name) of the built
 #'  package
 build <- function(path = ".", dest_path = NULL, binary = FALSE, vignettes = TRUE,
                   manual = FALSE, args = NULL, quiet = FALSE,
-                  needs_compilation = NA, compile_attributes = TRUE) {
+                  needs_compilation = pkg_has_src(path), compile_attributes = FALSE,
```

Comment:
I changed the default of `compile_attributes()` to `FALSE`.

Now that we are using `build()` in `remotes` it seems safer to not compile the attributes by default, presumably they are already working in the committed state so we shouldn't try to recompile them.

Reply:
That makes sense to me